### PR TITLE
Remove stdune from configurator and switch to csexp

### PR DESCRIPTION
**Pull request by rgrinberg**
- [x] Helper functions now exist in Import
- [x] I've made ocaml_config not export any stdune stuff
- [x] Conversion is done using csexp
- [x] vendor ocaml_config parsing in configurator.

@diml I'm wondering, perhaps we should just merge ocaml_config and configurator?

### CODE REVIEW

File: otherlibs/configurator/src/import.ml [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+  let write_lines ?binary fn lines =
+    with_file_out ?binary fn ~f:(fun oc ->
+        List.iter
+          ~f:(fun line ->
+            output_string oc line;
+            output_string oc "\n")
+          lines)
+
+  let read_all =
```
Comment (diml) [inline otherlibs/configurator/src/import.ml:235]: Note that we could use the simple version here. We only use `read_file` to read temporary files we created ourselves in configurator, so the simple version will always be enough.

**Comment by diml** [comment]
For ocaml-config, if we want to make it public I feel like it deserves its own repo. The feature it provides is well defined and makes sense on its own.

> I've made ocaml_config not export any stdune stuff

Why this change actually? If we still use `Stdune` in the implementation, then that's not really going to help us here.

BTW, the part of `Ocaml_config` that we use in configurator is very small. In this case, it would seem fine to me to copy it in configurator until we extract `ocaml-config` and make it public.

**Comment by rgrinberg** [comment]
> Why this change actually? If we still use Stdune in the implementation, then that's not really going to help us here.

It's just a first step. The next step is to get rid of stdune in the implementation as well. But without this, it wouldn't build.

> BTW, the part of Ocaml_config that we use in configurator is very small. In this case, it would seem fine to me to copy it in configurator until we extract ocaml-config and make it public.

Alright, let's do this then.

**Comment by rgrinberg** [comment]
@diml the ocaml_config dependency has been dropped from configurator. This is now ready for review.

File: otherlibs/configurator/src/import.ml (hunk 1 above)
Comment (rgrinberg) [reply otherlibs/configurator/src/import.ml:257]: That's true but I find that just copying the implementation directly from stdune makes maintenance easier. We can just copy/paste to update the code without worrying about little details

Comment (jeremiedimino) [reply otherlibs/configurator/src/import.ml:257]: That's true. BTW, the Io module (at least the String version) feels like something that could easily be cleaned up and stabilised. So we could mint some library for it to share the code.

**Pull request merged**